compiler/rustc_data_structures/src/marker.rs

```diff
@@ -0,0 +1,256 @@
+cfg_if!(
+    if #[cfg(not(parallel_compiler))] {
+        pub auto trait DynSend {}
+        pub auto trait DynSync {}
+
+        impl<T> DynSend for T {}
+        impl<T> DynSync for T {}
+    } else {
+        #[rustc_on_unimplemented(
+            message = "`{Self}` doesn't implement `DynSend`. \
+            Add it to `rustc_data_structures::marker` or use `IntoDyn` if it's already `Send`",
+            label = "`{Self}` doesn't implement `DynSend`. \
+            Add it to `rustc_data_structures::marker` or use `IntoDyn` if it's already `Send`"
+        )]
+        // Ensure data structures is `Send` by using runtime check.
+        // Note: Ensure that the data structure **will not break**
+        // thread safety after being created.
```

Comment:
This should mention that `Self: Send` if `sync::active()`. Same for `DynSync`.

Reply:
Yea I optimized this description